src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java

```diff
@@ -69,11 +70,18 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer e
         }
         return true;
       }
+      if (FluidUtil.fillInternalTankFromPlayerHandItem(world, x, y, z, entityPlayer, tank)) {
```

Comment:
But there is already code for that, see getInputTank and getOutputTanks in TileReservoir